pkg/config/reload.go

```diff
@@ -176,6 +177,15 @@ func (c *Config) ReloadDecryptionKeyConfig(newConfig *Config) {
 	}
 }
 
+// ReloadEnableHeapDump updates the DecryptionKeysPath with the provided
+// `newConfig`.
+func (c *Config) ReloadEnableHeapDump(newConfig *Config) {
```

Comment:
@hernandanielg, wasn't this meant to be a command-line argument only?

Reply:
The rationale behind this logic is to allow the user to change the location of the heap dump by reloading the config. So, we have to change the config option to a string type, rather than a boolean.

Reply:
generally, if a malicious user has access to the crio socket, they can do harm in a lot of different ways. admins should ensure only trusted users can access nodes. For instance, a user could DOS CRI-O by having systemd shut it off. Since this and the goroutine profiles are both debugging options, I think it's safe to guess a user who has access can be trusted to not DOS the node's memory

Reply:
Sorry I was confused, I thought this was only a command line option to generate it as it could be done by sending `SIGUSR1` to CRI-O engine. what do you think? it should be an endpoint?

Reply:
@hernandanielg, we've had some internal conversations about how to move forward with this feature.

There is, of course, still an opportunity to complete this work here, so please be sure to bear with us a little bit. :smile: 

That said, I would like to bring everyone's attention to the following discussion:

- https://github.com/cri-o/cri-o/discussions/7615

Reply:
hey any news about this?

Reply:
@hernandanielg, sorry for the delay!

A decision on how to proceed has stalled, so it seems. Apologies!